iceoryx_utils/test/moduletests/test_posix_mutex.cpp

```diff
@@ -47,7 +48,7 @@ class Mutex_test : public Test
         }
     }
 
-    iox::posix::mutex sut{false};
+    mutex_t sut{mutex_t::Recursive::OFF, mutex_t::Robust::OFF};
```

Comment:
Test cases for recursive and robust mutexes are missing.
Like:
- normal creation and usage should work for recursive, robust and recursive and robust mutex
- multiple locks should not lead to deadlock in a recursive mutex in a single thread
- multiple locks should cause a deadlock in a recursive mutex from different threads (start multiple threads, let the first thread deadlock and then unlock the mutex in the thread it was locked before)
- robust mutex cleanup should work ... here we maybe have to think how we can test it. @marthtz do you have an idea?